week-4/Homework/mandatory/4-groceries.js

```diff
@@ -1,12 +1,84 @@
-// You're going shopping, and you need a shopping list.
-// 1. Update your groceryList with the items you need: Potatoes, Orange Juice and Rice.
-// 2. Loop through the groceryList object to gather the item properties into the groceriesToBuy array.
-// 3. Then use console.log() to print out the list. It should print ['Potatoes', 'Orange Juice', 'Rice']
+/*
+As you you can have an Array of Objects, you can also store Arrays in Objects.
+In this exercise, you'll practice:
+ - How to loop through the properties (keys) of an Object and read its values.
+ - How to access Array stored inside Object.
+ - How to access to a specific property of an array and set it.
 
-let groceriesToBuy = [];
+You're going shopping, and you need a shopping list. You've already created your weekly meal plan
+that contains the missing ingredients to your menus. It is stored in the "weeklyMealPlan" object.
 
-let groceryList = {
-  item1: "",
-  item2: "",
-  item3: ""
+Complete the exercises below.
+*/
+
+// Here is your
+let weeklyMealPlan = {
+  monday: ["Cheese", "Eggs", "Tomato", "Paprika", "Leek"],
+  tuesday: ["Wrap", "Tuna", "Canned beans", "Cheese", "Carrot", "Aubergine"],
+  wednesday: ["Orange Juice", "Apple", "Ananas", "Black tea"],
+  thursday: ["Lamb", "Salt", "Bulgur", "Potato"],
+  friday: ["Rice milk", "Blueberries", "Porridge", "Banana", "Cinnamon"],
+  saturday: ["Olive oil", "Potato", "Salmon", "Asparagus"],
+  sunday: [],
 };
+
+/*
+Exercise 1:
+  Loop through the weekly meal plan object to gather weakly ingredients into the weeklyGroceriesToBuy array.
+  Then use console.log() to print out the list.
+*/
+// Gather all week item names into this array
+let weeklyGroceriesToBuy = Object.values(weeklyMealPlan);
+let weeklyItems = [];
+for (let i = 0; i < weeklyGroceriesToBuy.length; i++) {
+  for (let j = 0; j < weeklyGroceriesToBuy[i].length; j++) {
+    weeklyItems.push(weeklyGroceriesToBuy[i][j]);
+  }
+}
```

Comment:
This is actually working fine but you have other option to use such as for in loop, make it as an array then use methods.